docs/reference/wasm.md

```diff
@@ -0,0 +1,241 @@
+- `name`(string): Field's name
+
+Return value:
+- (u32): Field handle on success, 0 on error.
+
+#### `dataSourceAddField(u32 ds, string name, u32 kind) u32`
+
+Add a field to a data source
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `name`(string): Field's name
+- `kind` (u32): Field's kind. See values in https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget@main/pkg/gadget-service/api#Kind.
+
+Return value:
+- (u32): Field handle on success, 0 on error.
+
+#### `dataSourceNewPacketSingle(u32 ds) u32`
+
+Allocate a packet instance. The returned packet has to be released with
+`dataSourceEmitAndRelease` or `dataSourceRelease`.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+
+Return value:
+- (u32): Packet handle on success, 0 on error
+
+#### `dataSourceNewPacketArray(u32 ds) u32`
+
+Allocate a packet array instance. The returned packet has to be released with
+`dataSourceEmitAndRelease` or `dataSourceRelease`.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+
+Return value:
+- (u32): Packet handle on success, 0 on error
+
+#### `dataSourceEmitAndRelease(u32 ds, u32 data) u32`
+
+Emit and release a packet instance.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `data` (u32): Packet handle (as returned by `dataSourceNewPacketSingle`)
+
+Return value:
+- 0 in case of success, 1 otherwise.
+
+#### `dataSourceRelease(u32 ds, u32 data)`
+
+Release a packet instance without sending it.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `data` (u32): Packet handle (as returned by `dataSourceNewPacketSingle`)
+
+Return value:
+- None
+
+#### `dataArrayNew(d uint32) uint32`
+
+Allocate and returne a new element on the array.
+
+Parameters:
+- `d` (u32): DataArray handle
+
+Return value:
+- (u32): Data handle on success, 0 on error
+
+#### `dataArrayAppend(d uint32, data uint32)`
+
+Append data to the array.
+
+Parameters:
+- `d` (u32): DataArray handle
+- `data` (u32): Data handle to append
+
+Return value:
+- None
+
+#### `dataArrayRelease(d uint32, data uint32)`
+
+Releases the memory of Data; Data may not be used after calling this.
+
+Parameters:
+- `d` (u32): DataArray handle
+- `data` (u32): Data handle to release
+
+Return value:
+- None
+
+#### `dataArrayLen(d uint32) uint32`
+
+Get the number of elements in the array.
+
+Parameters:
+- `d` (u32): DataArray handle
+-
+Return value:
+- (u32): Number of elements in the array
+
+#### `dataArrayGet(d uint32, index uint32) uint32`
+
+Get the element at the given index.
+
+Parameters:
+- `d` (u32): DataArray handle
+- `index` (u32): Data index
+
+Return value:
+- (u32): Data handle on success, 0 on error
+
+### Fields
+
+#### `fieldAccessorGet(u32 acc, u32 data, u32 kind) u64`
```

Comment:
We're not using `accessor` in any other function name - maybe drop it to `fieldGet` or even call this one `fieldGetContent` as it doesn't have direct counterparts on the Go side and makes the function clearer.

Reply:
Renamed it to `field{Get,Set}`, I think that's better than having content or accessor in the names.